File: RQ3/Java-REval-Desc/task06_desc.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    public static List<Integer> rollingMax(List<Integer> numbers) {
        Integer runningMax = null;
        List<Integer> result = new ArrayList<>();

        for (Integer n : numbers) {
            if (runningMax == null) {
                runningMax = n;
            } else {
                runningMax = Math.max(runningMax, n);
            }

            result.add(runningMax);
        }

        List<Integer> _result = result;

        assert func(_result);  // POST: func is equivalent to: From a given list of integers, generate a list of rolling maximum element found until given moment in the sequence. >>> rolling_max([1, 2, 3, 2, 3, 4, 2]) [1, 2, 3, 3, 3, 4, 4]
    }
}

File: RQ3/C-REval-Desc/task06_desc.c
#include <stdio.h>
#include <stdlib.h>

int* rolling_max(int* numbers, int length) {
    int* result = (int*)malloc(length * sizeof(int));
    int running_max = -2147483648; // Initialize to the smallest int

    for (int i = 0; i < length; i++) {
        if (i == 0) {
            running_max = numbers[i];
        } else {
            if (running_max < numbers[i]) {
                running_max = numbers[i];
            }
        }
        result[i] = running_max;
    }

    return result;
}